From 275b262e85f3dcbc30d444032712ff5132b1e422 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Thu, 4 Sep 2014 13:42:20 +0200
Subject: [PATCH 051/117] customize: fix locking accounts with passwords

When setting the password for a locked account, make sure to still write
the password after the "!!" marker, otherwise the account will have no
password.
(cherry picked from commit ba65af34357eb79e37480758416cdc4e184689aa)
---
 mllib/password.ml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mllib/password.ml b/mllib/password.ml
index 6527138..8a5b6e0 100644
--- a/mllib/password.ml
+++ b/mllib/password.ml
@@ -109,13 +109,13 @@ let rec set_linux_passwords ~prog ?password_crypto g root passwords =
             match selector with
             | { pw_locked = locked;
                 pw_password = Password password } ->
-              if locked then "!!" else "" ^ encrypt password crypto
+              (if locked then "!!" else "") ^ encrypt password crypto
             | { pw_locked = locked;
                 pw_password = Random_password } ->
               let password = make_random_password () in
               printf (f_"Setting random password of %s to %s\n%!")
                 user password;
-              if locked then "!!" else "" ^ encrypt password crypto
+              (if locked then "!!" else "") ^ encrypt password crypto
             | { pw_locked = true; pw_password = Disabled_password } -> "!!*"
             | { pw_locked = false; pw_password = Disabled_password } -> "*" in
           user ^ ":" ^ pwfield ^ rest
-- 
2.5.0

